medusa/indexers/tvmaze/tvmaze_api.py

```diff
@@ -0,0 +1,816 @@
+                    v = self._cleanData(v)
+            self._setShowData(sid, k, v)
+
+        # get episode data
+        if getEpInfo:
+            # Parse banners
+            if self.config['banners_enabled']:
+                self._parseBanners(sid)
+
+            # Parse actors
+            if self.config['actors_enabled']:
+                self._parseActors(sid)
+
+            # Parse episode data
+            epsEt = self._get_episode_list(sid, specials=False)
+
+            if not epsEt:
+                log().debug('Series results incomplete')
+                raise tvmaze_showincomplete('Show search returned incomplete results (cannot find complete show on TVmaze)')
+
+            if 'episode' not in epsEt:
+                return False
+
+            episodes = epsEt['episode']
+            if not isinstance(episodes, list):
+                episodes = [episodes]
+
+            for cur_ep in episodes:
+                if self.config['dvdorder']:
+                    log().debug('Using DVD ordering.')
+                    use_dvd = cur_ep['dvd_season'] != None and cur_ep['dvd_episodenumber'] != None
+                else:
+                    use_dvd = False
+
+                if use_dvd:
+                    seasnum, epno = cur_ep.get('dvd_season'), cur_ep.get('dvd_episodenumber')
+                else:
+                    seasnum, epno = cur_ep.get('seasonnumber'), cur_ep.get('episodenumber')
+
+                if seasnum is None or epno is None:
+                    log().warning('An episode has incomplete season/episode number (season: %r, episode: %r)', seasnum, epno)
+                    continue  # Skip to next episode
+
+                # float() is because https://github.com/dbr/tvnamer/issues/95 - should probably be fixed in TVDB data
+                seas_no = int(float(seasnum))
+                ep_no = int(float(epno))
+
+                for k, v in cur_ep.items():
+                    k = k.lower()
+
+                    if v is not None:
+                        if k == 'filename':
+                            v = self.config['url_artworkPrefix'] % (v)
+                        else:
+                            v = self._cleanData(v)
+
+                    self._setItem(sid, seas_no, ep_no, k, v)
+
+        return True
+
+    def __getitem__(self, key):
+        """Handles tvmaze_instance['seriesname'] calls.
+        The dict index should be the show id
+        """
+        if isinstance(key, (int, long)):
+            # Item is integer, treat as show id
+            if key not in self.shows:
+                self._getShowData(key, self.config['language'], True)
+            return self.shows[key]
+
+        key = str(key).lower()
+        self.config['searchterm'] = key
+        selected_series = self._getSeries(key)
+        if isinstance(selected_series, dict):
+            selected_series = [selected_series]
+        [[self._setShowData(show['id'], k, v) for k, v in show.items()] for show in selected_series]
```

Comment:
I think I can resolv this with _ = [[self._setShowData..., but would that be a proper solution?